### ISSUE TITLE
Maximum execution time on Header block

### ISSUE THREAD

**Issue by Yonier** [Textalk/websocket-php#156]
**Describe the bug**
When the library can't connect to the Websocket server, keeps doing the `do while` and as response i get a php error:

```
Severity: Error

Message: Maximum execution time of 30 seconds exceeded

Filename: lib/Client.php

Line Number: 183

Backtrace:
```

This was working well before the #154 commit

As a temporary fix I replaced the new block for the:
`$response = stream_get_line($this->socket, 1024, "\r\n\r\n");`

So when the library can't connect the rest of the code won't stop. Hope this can be fixed.
Thanks.

**Versions**
* Version of this library: 1.5
* PHP version: 8.0

**Comment by sirn-se** [comment]
I suppose we need to check if `fgets()` return false and abort here. I will look into it when I have some time to test it.

**Comment by sirn-se** [comment]
Fixed in `v1.5.8`, closing.
